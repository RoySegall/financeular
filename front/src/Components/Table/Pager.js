import {Next, Prev} from "../Icons/Icons";
import React from "react";

export default ({prevPageCallback, currentPage, numberOfPages, nextPageCallback}) => <span>
  <button onClick={prevPageCallback} className="pr-1 cursor-pointer underline"><Prev/></button>
  {currentPage + 1} of {numberOfPages}
  <button onClick={nextPageCallback} className="pl-1 cursor-pointer underline"><Next/></button>
</span>
