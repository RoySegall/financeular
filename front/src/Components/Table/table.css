.table-wrapper {
    width: 100%;
}

.table-wrapper .table {
    width: 100%;
    margin: 0 auto;
    border: #89c3f5 solid 1px;

}

.table-wrapper .table td {
    padding: .25em;
    margin-top: .25em;
    border: #89c3f5 solid 1px;
}

.table-wrapper thead td {
    background: #afe0f6;
    color: #3f7189;
}

.table-wrapper tbody tr:nth-child(odd) td {
    background: #ffffff;
    color: #0062a8;
}

.table-wrapper tbody tr:nth-child(even) td {
    background: #f1f1f1;
    color: #22445c;
}
