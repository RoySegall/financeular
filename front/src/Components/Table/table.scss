.table-wrapper {
    width: 100%;

    .table {
        width: 100%;
        margin: 0 auto;
        border: #89c3f5 solid 1px;

        td {
            padding: .25em;
            margin-top: .25em;
            border: #89c3f5 solid 1px;
        }

        thead td {
            background: #afe0f6;
            color: #3f7189;
        }

        tbody {
            tr:nth-child(odd) td {
                background: #ffffff;
                color: #0062a8;
            }

            tr:nth-child(even) td {
                background: #f1f1f1;
                color: #22445c;
            }
        }
    }
}
