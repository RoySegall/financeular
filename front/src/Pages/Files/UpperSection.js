import {formatToCurrency} from "./File.service";
import React from "react";

export default ({infoBoxTitle, infoBoxColor, infoBoxIcon, balance, setInfoMode, totalExpenses, totalIncomes, width}) => <section id="upper-wrapper" className="flex flex-row justify-between">
  <section id="balance-intro" className="pl-2 flex items-center">

    <div className="icon pr-2">
      <div className={`m-auto m-0 p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full text-2xl bg-${infoBoxColor}-400 text-white`}>
        {infoBoxIcon}
      </div>
    </div>
    <div className="icon align-center">
      <p className="text-xl font-bold ">{infoBoxTitle}</p>
      <p>
        The balance for the current month is <b>{formatToCurrency(balance)}</b>.
      </p>
    </div>
  </section>

  <section id="balance-info" className="pr-2 flex flex-col">

    <div className="flex">
      <div className="text-red-300">
        <span className="text-xl font-bold block"><button className="font-bold underline pointer-cursor" onClick={() => setInfoMode('expenses')}>Expenses</button></span>
        <span>{formatToCurrency(totalExpenses)}</span>
      </div>

      <div className="pl-8 text-green-500">
        <span className="text-xl block"><button className='font-bold underline pointer-cursor' onClick={() => setInfoMode('incomes')}>Incomes</button></span>
        <span>{formatToCurrency(totalIncomes)}</span>
      </div>
    </div>

    <div className="relative w-full" style={{maxWidth: '200px'}}>
      <div className="m-0 m-auto mt-2 h-6 w-full bg-green-200 rounded">
        &nbsp;
      </div>

      <div className="absolute top-0 m-0 m-auto mt-2 h-6 bg-red-200 rounded text-red-700 font-bold text-center" style={{width: (width * 2) + 'px', maxWidth: '210px'}}>
        {width}%
      </div>
    </div>

  </section>

</section>
