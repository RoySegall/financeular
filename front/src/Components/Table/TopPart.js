import Pager from "./Pager";
import React from "react";

export default ({title, actions=null, showPager, currentPage, numberOfPages, setCurrentPage}) => <div className="rounded-t mb-0 px-4 py-3 border-0">
  <div className="flex flex-wrap items-center">
    <div className="relative w-full px-4 max-w-full flex-grow flex text-left flex justify-between">
      <h3 className={"font-semibold text-lg text-gray-800"}>{title}</h3>
      {actions}

      {showPager && <Pager
        prevPageCallback={() => {currentPage > 0 && setCurrentPage(currentPage - 1)}}
        currentPage={currentPage}
        numberOfPages={numberOfPages}
        onClick1={() => {currentPage + 1 < numberOfPages && setCurrentPage(currentPage + 1)}
        }/>}
    </div>
  </div>
</div>
