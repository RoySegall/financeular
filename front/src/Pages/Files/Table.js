import {formatToCurrency} from "./File.service";
import React from "react";
import Pager from "./Pager";

export default ({tableHeaders, borderColor, rows, tableTdWidth, perPage, currentPage, setCurrentPage, infoMode}) => {

  const showPager = rows.length > perPage;

  let numberOfPages = null;

  if (showPager) {
    numberOfPages = Math.ceil(rows.length / perPage);
  }

  const rowsToDisplay = !showPager ? rows : rows.slice(perPage * currentPage, perPage * (currentPage + 1));

  return <table className="w-full">

    <thead>
    {tableHeaders.map((tableHeader, key) => <th
      className={"text-left font-semibold uppercase py-3 px-2 border border-solid border-l-0 border-r-0 border-gray-200 bg-gray-100 text-gray-600"}
      key={key}>
      {tableHeader}
    </th>)}
    </thead>

    <tbody>

    {rowsToDisplay.map((rows, key) => {
      return <tr className="text-black border-b border-gray-300" key={key}>
        {rows.map((item, tdKey) => <td className={`px-2 ${tableTdWidth} pt-2`} key={tdKey}>{tdKey === 1 ? formatToCurrency(item) : item}</td>)}
        {infoMode === 'expenses' && <td colSpan={2}><p className="bg-green-100 p-2 border border-green-900 w-min text-black my-2 font-light text-xs rounded">Category</p></td>}
      </tr>
    })}

    {numberOfPages > 1 && <tr>
      <td colSpan={5} className="pr-2 pt-4">
        <Pager
          pages={numberOfPages}
          borderColor={borderColor}
          currentPage={currentPage}
          setCurrentPage={setCurrentPage}
          numberOfPages={numberOfPages} />
      </td>
    </tr>}

    </tbody>
  </table>
}