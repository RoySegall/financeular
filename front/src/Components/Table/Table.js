import React from "react";
import './table.css'
import {Shekel} from "../Icons/Icons";

export default () => <div className="table-wrapper">
    <table className="table">
        <thead>
            <tr className="headers">
                <td>Title</td>
                <td>Date</td>
                <td>Amount</td>
            </tr>
        </thead>

        <tbody>
            <tr>
                <td>Burger</td>
                <td>08/01</td>
                <td>250 <Shekel /></td>
            </tr>
            <tr>
                <td>Burger</td>
                <td>08/01</td>
                <td>200 <Shekel /></td>
            </tr>
            <tr>
                <td>Burger</td>
                <td>08/01</td>
                <td>200 <Shekel /></td>
            </tr>
        </tbody>
    </table>

</div>
