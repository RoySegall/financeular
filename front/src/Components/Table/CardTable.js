import React, {useState} from "react";
import Table from "./Table";
import TopPart from "./TopPart";

export default function CardTable({ title, headers, rows, className, perPage, actions = null }) {
  const showPager = rows.length > perPage;
  const [currentPage, setCurrentPage] = useState(0);

  let numberOfPages = null;

  if (showPager) {
    numberOfPages = Math.ceil(rows.length / perPage);
  }

  const rowsToDisplay = !showPager ? rows : rows.slice(perPage*currentPage, perPage * (currentPage + 1));

  return <div className={`relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded bg-white ${className}`}>
    <TopPart title={title} actions={actions} showPager={showPager} currentPage={currentPage} setCurrentPage={setCurrentPage} numberOfPages={numberOfPages} />
    <div className="block w-full overflow-x-auto">
      <Table headers={headers} rows={rowsToDisplay} />
    </div>
  </div>;
}
