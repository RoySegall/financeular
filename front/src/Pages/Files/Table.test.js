import renderer from "react-test-renderer";
import UpperSection from "./UpperSection";
import React from "react";
import Table from "./Table";

describe('Table component', () => {

  it('Snap shot the component', () => {
    const tree = renderer
      .create(<Table
        tableHeaders={['title', 'value']}
        borderColor={'green'}
        rows={[[1,2], [2,1]]}
        tableTdWidth={2}
        perPage={1}
        currentPage={0}
        setCurrentPage={() => {}}
        infoMode={'expenses'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
