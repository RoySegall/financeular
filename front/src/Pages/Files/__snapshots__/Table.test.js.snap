// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Snap shot the component 1`] = `
<table
  className="w-full"
>
  <thead>
    <th
      className="text-left font-semibold uppercase py-3 px-2 border border-solid border-l-0 border-r-0 border-gray-200 bg-gray-100 text-gray-600"
    >
      title
    </th>
    <th
      className="text-left font-semibold uppercase py-3 px-2 border border-solid border-l-0 border-r-0 border-gray-200 bg-gray-100 text-gray-600"
    >
      value
    </th>
  </thead>
  <tbody>
    <tr
      className="text-black border-b border-gray-300"
    >
      <td
        className="px-2 2 pt-2"
      >
        1
      </td>
      <td
        className="px-2 2 pt-2"
      >
        2
      </td>
      <td
        colSpan={2}
      >
        <p
          className="bg-green-100 p-2 border border-green-900 w-min text-black my-2 font-light text-xs rounded"
        >
          Category
        </p>
      </td>
    </tr>
    <tr>
      <td
        className="pr-2 pt-4"
        colSpan={5}
      >
        <ul
          className="flex float-right"
        >
          <li
            className="p-2 border border-green-400 bg-green-50 text-green-700 border-r-0 rounded-l-lg"
          >
            <button
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-angle-double-left fa-w-14 "
                data-icon="angle-double-left"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
          </li>
          <li
            className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
          >
            <button
              className="font-semibold text-black"
              onClick={[Function]}
            >
              1
            </button>
          </li>
          <li
            className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
          >
            <button
              className="text-green-700"
              onClick={[Function]}
            >
              2
            </button>
          </li>
          <li
            className="p-2 border border-green-400 bg-green-50 text-green-700 rounded-r-lg"
          >
            <button
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-angle-double-right fa-w-14 "
                data-icon="angle-double-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
          </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>
`;
